pkg/kudoctl/kudoinit/migration/01_migrate_ov_names/migrate_ov_names.go

```diff
@@ -0,0 +1,119 @@
+//nolint:golint,stylecheck
+package _01_migrate_ov_names
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kudoapi "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kube"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kudoinit/migration"
+)
+
+var (
+	_ migration.Migrater = &MigrateOvNames{}
+)
+
+type MigrateOvNames struct {
+	client *kube.Client
+	dryRun bool
+	ctx    context.Context
+
+	migratedOVs map[string]*kudoapi.OperatorVersion
+}
+
+func New(client *kube.Client, dryRun bool) *MigrateOvNames {
+	return &MigrateOvNames{
+		client:      client,
+		dryRun:      dryRun,
+		ctx:         context.TODO(),
+		migratedOVs: map[string]*kudoapi.OperatorVersion{},
+	}
+}
+
+func (m *MigrateOvNames) CanMigrate() error {
+	// No migrate check required for this migration
+	return nil
+}
+
+func (m *MigrateOvNames) Migrate() error {
+	if !m.dryRun {
+		clog.V(0).Printf("Migrate OperatorVersion names")
+	}
+	return migration.ForEachNamespace(m.client, m.migrateInNamespace)
+}
+
+func (m *MigrateOvNames) migrateInNamespace(ns string) error {
+	clog.V(1).Printf("Run OperatorVersion name migration on namespace %q", ns)
+	if err := migration.ForEachOperatorVersion(m.client, ns, m.migrateOperatorVersion); err != nil {
```

Comment:
this is not really atomic because before you migrate instance, it has no OV linked or the link is rather broken :/

Reply:
so what if this fails in the middle? we'll leave the cluster in broken state with no easy way to fix this :/

Reply:
maybe the re-run of upgrade would actually fix itself? 🤔 but will it even work if manager is down and everything is "in between"?

Reply:
Yeah - the migration could have been fixed to be completely reentrant, the code that you reviewed wasn't though. I've removed the migration now, but we should keep that in mind for future migrations.